src/mlpack/methods/adaboost/adaboost_main.cpp

```diff
@@ -226,23 +226,32 @@ static void mlpackMain()
     Timer::Start("adaboost_training");
     m->Train(trainingData, labels, numClasses, iterations, tolerance);
     Timer::Stop("adaboost_training");
+    if(IO::HasParam("test"))
+    {
+      mat testingData = std::move(IO::GetParam<arma::mat>("test"));
+      CheckInputShape(trainingData.n_rows, trainingData.n_cols, labels.n_elem, testingData.n_rows);
+    }
+    else
+      CheckInputShape(trainingData.n_rows, trainingData.n_cols, labels.n_elem, 0);
   }
   else
   {
     // We have a specified input model.
     m = IO::GetParam<AdaBoostModel*>("input_model");
   }
 
+  
+
   // Perform classification, if desired.
   if (IO::HasParam("test"))
   {
     mat testingData = std::move(IO::GetParam<arma::mat>("test"));
 
-    if (testingData.n_rows != m->Dimensionality())
+    if (testingData.n_rows != m->Dimensionality() && testingData.n_cols > 0)
```

Comment:
The general idea would be to use a utility function here instead of a handwritten check.